ios/ios/src/Web Services/CMAPICredentials.m

```diff
@@ -31,6 +32,8 @@ - (void)setAppIdentifier:(NSString *)appId apiKey:(NSString *)apiKey andBaseURL:
     self.appIdentifier = appId;
     self.apiKey = apiKey;
     self.baseURL = baseURL;
+
+    [CMLegacyCacheCleaner cleanLegacyCache];
```

Comment:
Agree it's odd, but after thinking through it, I think it's the best option. Thought about the App Delegate, but we don't (necessarily) control that class: unless you're using push notifications, you don't need to subclass our AppDelegate.

My thought is most customers will be performing this call in their AppDelegate didFinishLaunching: method. One thing I guess I hadn't considered, and is just now occurring to me, is that they might just set these properties manually, instead of using the method.

Do you have any ideas for a better spot, @marcweil?